jstests/core/hidden_index.js

```diff
@@ -0,0 +1,113 @@
+	assert(usedIndex(query));
+	
+	assert.commandWorked(testtab.hideIndex(index_name));
+	assert(testtab.getIndexes()[1].hidden);
+	if (index_type === "text") {
+	    // errmsg: error processing query: ns=test.hidden_indexTree: TEXT : query=java,
+	    // language=english, caseSensitive=0, diacriticSensitive=0,
+	    // tag=NULL\nSort: {}\nProj: {}\n planner returned error ::
+	    // caused by :: need exactly one text index for $text query
+	    printjson(query);
+	    assert.throws(() => {testtab.find(query).explain();});
+	    return;
+	}
+	assert(!usedIndex(query));
+	
+	assert.commandWorked(testtab.unhideIndex(index_name));
+	assert.eq(testtab.getIndexes()[1].hidden, undefined);
+	assert(usedIndex(query));
+	
+	assert.commandWorked(testtab.dropIndex(index_name));
+
+	if (wildcard)
+	    assert.commandWorked(testtab.createIndex({"a.$**": index_type}, {hidden: true}));
+	else
+	    assert.commandWorked(testtab.createIndex({"a": index_type}, {hidden: true}));
+
+	assert(testtab.getIndexes()[1].hidden);
+	assert(!usedIndex(query));
+    }
+
+    
+    // normal index
+    HiddenCheckTest({a: 1}, 1);
+    // geo index
+    HiddenCheckTest({a: { $geoWithin :
+			  { $geometry :
+			    { type : "Polygon" ,
+                              coordinates : [ [
+                                  [ 0 , 0 ] ,
+                                  [ 3 , 6 ] ,
+                                  [ 6 , 1 ] ,
+                                  [ 0 , 0 ]
+                              ] ]
+			    } } } }, "2dsphere");
+
+    // fts index
+    HiddenCheckTest({ $text: { $search: "java" } }, "text");
+
+    // wildcard index
+    HiddenCheckTest({"a.f": 1}, 1, true);
+
+    // hidden index on capped collection
+    assert(testtab.drop());
+    assert.commandWorked(db.createCollection(jsTest.name(), {capped: true, size: 100}));
+    HiddenCheckTest({a: 1}, 1);
+    
+    // can't hide index on system index
+    const systemtab = db.getSiblingDB('admin').system.version;
+    assert.commandWorked(systemtab.createIndex({a: 1}));
+    assert.commandFailed(systemtab.hideIndex("a_1"));
+    assert.commandWorked(systemtab.dropIndex("a_1"));
+    assert.commandFailed(systemtab.createIndex({a: 1}, {hidden: true}));
+
+    // can't hide _id index
+    assert.commandFailed(testtab.hideIndex("_id_"));
+
+    // we can change ttl index and hide info at the same timex
+    assert.commandWorked(testtab.dropIndexes());
+    assert.commandWorked(testtab.createIndex({"tm": 1}, {expireAfterSeconds: 10}));
+    assert.eq(testtab.getIndexes()[1].hidden, undefined);
+    assert.eq(testtab.getIndexes()[1].expireAfterSeconds, 10);
+
+    db.runCommand({"collMod": testtab.getName(), "index": {"name": "tm_1",
+							   "expireAfterSeconds": 1,
+							   "hidden": true
+							  }});
+    assert(testtab.getIndexes()[1].hidden);
+    assert.eq(testtab.getIndexes()[1].expireAfterSeconds, 1);
+
+    // after we hide the index, the ttl still work as well
```

Comment:
Was this comment meant to be paired with an additional test? I think confirming that documents are deleted via TTL on a hidden index should be a separate test and probably one with its own test fixture. An example of a similar test is: jstests/noPassthrough/ttl_partial_index.js